java-frontend/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -125,37 +126,41 @@ private ProgramState(ProgramState ps, PMap<SymbolicValue, Constraint> newConstra
   }
 
   ProgramState stackValue(SymbolicValue sv) {
-    Deque<SymbolicValue> newStack = new LinkedList<>(stack);
-    newStack.push(sv);
-    return new ProgramState(this, newStack);
+    return new ProgramState(this, stack.push(sv));
   }
 
   ProgramState clearStack() {
-    return unstackValue(stack.size()).state;
+    return unstackValue(Integer.MAX_VALUE).state;
```

Comment:
I would prefer to have a dedicated method to handle this case rather than going for Integer.MAX_VALUE. Let's keep something meaningful. (or even -1 but let's bound the iteration properly).